redis/src/aio.rs

```diff
@@ -642,6 +642,7 @@ pin_project! {
         sink_stream: T,
         in_flight: VecDeque<InFlight<I, E>>,
         error: Option<E>,
+        redis_errors: VecDeque<E>,
```

Comment:
What's the distinction between `error` and `redis_errors` here? Having both of those seems... weird.

Reply:
Agreed. `error` basically indicates that `Sink` has failed, so it's basically a terminal state. I could rename it to `sink_error` or `terminal_error` to indicate that.

Reply:
I renamed both fields and made `redis_errors` an Option<E> named `first_err` to match the other Connection types. Both are needed though, as they serve different purposes.

Reply:
Moved mutable fields onto the Inflight entry, which is more appropriate here.